Content.Shared/Ensnaring/SharedEnsnareableSystem.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Content.Shared.Ensnaring.Components;
+using Content.Shared.Movement.Systems;
+
+namespace Content.Shared.Ensnaring;
+
+public abstract class SharedEnsnareableSystem : EntitySystem
+{
+    [Dependency] private readonly MovementSpeedModifierSystem _speedModifier = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SharedEnsnareableComponent, RefreshMovementSpeedModifiersEvent>(MovementSpeedModify);
+        SubscribeLocalEvent<SharedEnsnareableComponent, EnsnareEvent>(OnEnsnare);
+        SubscribeLocalEvent<SharedEnsnareableComponent, EnsnareRemoveEvent>(OnEnsnareRemove);
+        SubscribeLocalEvent<SharedEnsnareableComponent, EnsnaredChangedEvent>(OnEnsnareChange);
+    }
+
+    private void OnEnsnare(EntityUid uid, SharedEnsnareableComponent component, EnsnareEvent args)
+    {
+        component.WalkSpeed = args.WalkSpeed;
+        component.SprintSpeed = args.SprintSpeed;
+
+        _speedModifier.RefreshMovementSpeedModifiers(uid);
+
+        var ev = new EnsnaredChangedEvent(component.IsEnsnared);
+        RaiseLocalEvent(uid, ev, true);
+    }
+
+    private void OnEnsnareRemove(EntityUid uid, SharedEnsnareableComponent component, EnsnareRemoveEvent args)
+    {
+        _speedModifier.RefreshMovementSpeedModifiers(uid);
+
+        var ev = new EnsnaredChangedEvent(component.IsEnsnared);
+        RaiseLocalEvent(uid, ev, true);
```

Comment:
Are these used as broadcast events (passing in true)?

Reply:
They are so the stripping menu can pick it up, like with handcuffs.